win32_threads.c

```diff
@@ -2466,6 +2603,20 @@ GC_INNER void GC_thr_init(void)
     }
 # endif
 
+# ifdef WOW64_THREAD_CONTEXT_WORKAROUND
+    /* Set isWow64 flag. */
+    {
+      HMODULE hK32 = GetModuleHandle(TEXT("kernel32.dll"));
+      if (hK32) {
+        FARPROC pfn = GetProcAddress(hK32, "IsWow64Process");
+        if (pfn
+            && !(*(BOOL (WINAPI*)(HANDLE, BOOL*))pfn)(GetCurrentProcess(),
+                                                      &isWow64))
```

Comment:
1. We should call this directly. We don't support WinXP anymore;
2. Microsoft highly recommends calling `IsWow64Process2` if it's available, as `IsWow64Process` is a bit flawed.

Reply:
Looks like `IsWow64Process2` is Windows 10 minimum, should we dynamically lookup that then fallback to `IsWow64Process`?

Reply:
Yes, we actually started using it within Unity recently too:

```
auto wow64Module = GetModuleHandleW(L"api-ms-win-core-wow64-l1-1-1.dll");
if (wow64Module != nullptr)
{
    typedef BOOL(WINAPI* IsWow64Process2Func)(HANDLE hProcess, USHORT* pProcessMachine, USHORT* pNativeMachine);
    auto isWow64Process2 = reinterpret_cast<IsWow64Process2Func>(GetProcAddress(wow64Module, "IsWow64Process2"));
    if (isWow64Process2 != nullptr)
    {
        ....
```

Reply:
Oh, almost forgot: if you end up using it, you should forward declare `GetModuleHandleW` as while it's allowed on UWP, it is #fidef'd in the headers to "discourage" people from using it.

`extern "C" __declspec(dllimport) HMODULE WINAPI GetModuleHandleW(LPCWSTR lpModuleName);`

Reply:
(we can talk on slack about it as then I can link to our source code where we use it).

Reply:
Resolved in latest commit.